firefox_puppeteer/base.py

```diff
@@ -28,16 +28,44 @@ def get_marionette(self):
 class UIBaseLib(BaseLib):
     """A base class for all UI element wrapper classes inside a chrome window."""
 
-    def __init__(self, marionette_getter, window):
-        BaseLib.__init__(self, marionette_getter)
-
+    def __init__(self, marionette_getter, parent, element):
```

Comment:
I'm a little skeptical of adding "parent" to all our ui classes. Some elements don't have a parent, but moreover I think this opens us up to confusion by what's specified here and what's the actual parent element in the dom (which could probably get out of sync in some cases).